drivers/frequency/adf4382/iio_adf4382.c

```diff
@@ -0,0 +1,989 @@
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdlib.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "iio_adf4382.h"
+#include "adf4382.h"
+#include "no_os_alloc.h"
+
+/**
+ * @brief Supported charge pump currents
+ */
+static const int ci_table[16][2] = {
+	{0, 700000},
+	{0, 900000},
+	{1, 100000},
+	{1, 300000},
+	{1, 500000},
+	{1, 900000},
+	{2, 300000},
+	{2, 700000},
+	{3, 100000},
+	{3, 900000},
+	{4, 700000},
+	{5, 500000},
+	{6, 300000},
+	{7, 900000},
+	{9, 500000},
+	{11, 100000}
+};
+
+/**
+ * @brief Helper function to read a pair of values in a table with 2 elements.
+ * @param tb	- Input table.
+ * @param size	- Size of the table.
+ * @param val1	- The first value to be found.
+ * @param val2	- The second value to be found.
+ * @return i	- The index of the pair if found or an error if there is no
+ * 		  pair of values in the table.
+ */
+static int adf4382_iio_find_2d_row(const int (*tbl)[2], const int size,
```

Comment:
You may use `ci_table` directly in this function without needing to pass it as a parameter.